File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/create/NewMomentContract.java
package com.zxjdev.smile.presentation.moment.create;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;

public interface NewMomentContract {

  interface View extends BaseView {

    void close();
  }

  interface Presenter extends BasePresenter<View> {

    void handleAddMoment(String context);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/moment/create/NewMomentContract.kt
package com.zxjdev.smile.presentation.moment.create

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView

interface NewMomentContract {

    interface View : BaseView {

        fun close()
    }

    interface Presenter : BasePresenter<View> {

        fun handleAddMoment(content: String)
    }
}
